File: youtube-links/src/shared/features/users/commands/UpdateUserTheme.ts
import { isValidThemeColorValue, type ThemeColor } from '../helpers/ThemeColor';

export namespace UpdateUserTheme {
  export interface Command {
    id: number;
    themeColor: ThemeColor;
  }

  export const Validation = {
    themeColor: {
      isInEnum: (v: string) =>
        isValidThemeColorValue(v) ||
        `ThemeColor has a range of values which does not include: ${v}.`,
    },
  };
}

File: YoutubeLinks.Shared/Features/Users/Commands/UpdateUserTheme.cs
using FluentValidation;
using MediatR;
using Microsoft.Extensions.Localization;
using YoutubeLinks.Shared.Features.Users.Helpers;
using YoutubeLinks.Shared.Localization;

namespace YoutubeLinks.Shared.Features.Users.Commands
{
    public static class UpdateUserTheme
    {
        public class Command : IRequest<Unit>
        {
            public int Id { get; set; }
            public ThemeColor ThemeColor { get; set; }
        }

        public class Validator : AbstractValidator<Command>
        {
            public Validator(IStringLocalizer<ValidationMessage> localizer)
            {
                RuleFor(x => x.ThemeColor)
                    .IsInEnum()
                    .WithMessage(x => localizer[nameof(ValidationMessageString.ThemeColorIsInEnum)]);
            }
        }
    }
}
